ebi_eva_common_pyutils/config_utils.py

```diff
@@ -53,3 +56,40 @@ def get_mongo_uri_for_eva_profile(eva_profile_name: str, settings_xml_file: str)
     authentication_db = config.get_value_with_xpath(
         xpath_location_template.format(eva_profile_name, "eva.mongo.auth.db"))[0]
     return "mongodb://{0}:{1}@{2}/{3}".format(username, quote_plus(password), mongo_hosts_and_ports, authentication_db)
+
+
+def get_properties_from_xml_file(profile, xml_path):
+    tree = et.parse(xml_path)
+    root = tree.getroot()
+    return get_profile_properties(profile, root)
+
+
+def get_properties_from_xml_string(profile, str):
+    root = et.fromstring(str)
+    return get_profile_properties(profile, root)
+
+
+def get_profile_properties(profile, root):
+    properties = {}
+    for property in root.xpath('//settings/profiles/profile/id[text()="' + profile + '"]/../properties/*'):
+        properties[property.tag] = property.text
+    return properties
+
+
+def get_eva_settings_xml_string(token):
+    url = 'https://api.github.com/repos/EBIvariation/configuration/contents/eva-maven-settings.xml'
+    headers = {'Authorization': 'token ' + token, 'Accept' : 'application/vnd.github.raw' }
+    request = urllib.request.Request(url, None, headers)
```

Comment:
Suggest retry decorator for methods that do API accesses... Github is notorious for its downtime...